pkg/apis/core/validation/validation.go

```diff
@@ -2899,6 +2904,20 @@ func validateSecretKeySelector(s *core.SecretKeySelector, fldPath *field.Path) f
 	return allErrs
 }
 
+func validateFileKeySelector(s *core.FileKeySelector, fldPath *field.Path) field.ErrorList {
```

Comment:
What about all the other fields?  Need to ensure the path is specified and doesn't have anything sneaky like ".." in it, and that Optional is specified (the API says "optional" but it has a default, so it is effectively required.